src/realm/sync/subscriptions.cpp

```diff
@@ -0,0 +1,450 @@
+    }
+    return static_cast<State>(m_obj.get<int64_t>(m_mgr->m_sub_set_keys->state));
+}
+
+StringData SubscriptionSet::error_str() const
+{
+    if (!m_obj.is_valid()) {
+        return StringData{};
+    }
+    return m_obj.get<StringData>(m_mgr->m_sub_set_keys->error_str);
+}
+
+size_t SubscriptionSet::size() const
+{
+    if (!m_obj.is_valid()) {
+        return 0;
+    }
+    return m_sub_list.size();
+}
+
+SubscriptionSet::const_iterator SubscriptionSet::begin() const
+{
+    return iterator(this, m_sub_list.begin());
+}
+
+SubscriptionSet::const_iterator SubscriptionSet::end() const
+{
+    return iterator(this, m_sub_list.end());
+}
+
+SubscriptionSet::const_iterator SubscriptionSet::find(StringData name) const
+{
+    return std::find_if(begin(), end(), [&](const Subscription& sub) {
+        return sub.name() == name;
+    });
+}
+
+SubscriptionSet::const_iterator SubscriptionSet::find(const Query& query) const
+{
+    const auto query_desc = query.get_description();
+    const auto table_name = Group::table_name_to_class_name(query.get_table()->get_name());
+    return std::find_if(begin(), end(), [&](const Subscription& sub) {
+        return sub.object_class_name() == table_name && sub.query_string() == query_desc;
+    });
+}
+
+SubscriptionSet::iterator SubscriptionSet::begin()
+{
+    return iterator(this, m_sub_list.begin());
+}
+
+SubscriptionSet::iterator SubscriptionSet::end()
+{
+    return iterator(this, m_sub_list.end());
+}
+
+SubscriptionSet::iterator SubscriptionSet::erase(iterator it)
+{
+    m_sub_list.remove_target_row(it.m_sub_it.index());
+    return it;
+}
+
+void SubscriptionSet::clear()
+{
+    m_sub_list.remove_all_target_rows();
+}
+
+void SubscriptionSet::insert_sub_impl(Timestamp created_at, Timestamp updated_at, StringData name,
+                                      StringData object_class_name, StringData query_str)
+{
+    auto new_sub = m_sub_list.create_and_insert_linked_object(m_sub_list.is_empty() ? 0 : m_sub_list.size() - 1);
+    new_sub.set(m_mgr->m_sub_keys->created_at, created_at);
+    new_sub.set(m_mgr->m_sub_keys->updated_at, updated_at);
+    new_sub.set(m_mgr->m_sub_keys->name, name);
+    new_sub.set(m_mgr->m_sub_keys->object_class_name, object_class_name);
+    new_sub.set(m_mgr->m_sub_keys->query_str, query_str);
+}
+
+Subscription SubscriptionSet::subscription_from_iterator(LnkLst::iterator it) const
+{
+    if (it == m_sub_list.end()) {
+        return Subscription(this, Obj{});
+    }
+    return Subscription(this, m_sub_list.get_object(it.index()));
+}
+
+std::pair<SubscriptionSet::iterator, bool> SubscriptionSet::insert(const Query& query,
+                                                                   util::Optional<std::string> name)
+{
+    auto table_name = Group::table_name_to_class_name(query.get_table()->get_name());
+
+    auto query_str = query.get_description();
+    if (!name) {
+        name = util::format("%1: %2", table_name, query_str);
```

Comment:
Can we just set an empty string for the name instead? The query description is liable to change if the query changes. Also I think it disagrees with the comment about names:
```
// Returns the name of the subscription that was set when it was created. If no name was set, this
// returns a default-constructed std::string_view.
```

Reply:
Ahh, hmm. I think the comment is wrong. I will update the comment.